### ISSUE TITLE
Expose Chain.spend tx

### ISSUE THREAD

**Issue by mradkov** [aeternity/ae_mdw#24]
Receive tokens transactions originating from `Chain.spend()` are not indexed as inbound transactions for accounts.

E.g.
Alice (ak_123)
Bob (ak_456)

Flow:
Alice (ak_123) -> [contractCall tx] -> Contract method -> Chain.spend(ak_456) 

Bob does receive the tokens but the middleware is not indexing this so the outcome is -> balance in the account without history where it came from.

**Comment by marc0olo** [comment]
@mradkov @skkw what's the state in that regards? can we also cover the name_id resolution here?

we should also be able to list the SpendTx an account with a **_pointer at a certain point of time_** received. there could also the the following scenario:
```
Keyblock   | Type     | name_id  | pointer (account_pubkey)
---------------------------------------------------------------
123        | SpendTx  | nm_abc   | ak_aaa
345        | SpendTx  | nm_abc   | ak_bbb
567        | SpendTx  | nm_abc   | ak_ccc
```

so we have different transactions for the same name_id and in each transaction there is a different recipient. we also need to be able to fetch that information. of course this includes the `Chain.spend` usage within contracts, too

**Comment by skkw** [comment]
`Chain.spend` in contracts, and generally any calls in contracts needs to be implemented in AE node itself - at the FATE interpreter level.
Once/if node generates such information, the MDW - layer on top of AE node, can index that...

The naming system/auctions - I just started looking into this...
Pointers & name resolution - it should be doable to have convenient support in this are in 3-4 weeks.

**Comment by marc0olo** [comment]
ok, then I am looking forward to see that implemented in the node :-)

**Comment by davidyuk** [comment]
There is the same issue with tips claimed through a contract. It is a bit tricky to merge two lists (from the middleware and our backend) on the client-side to show the transaction list.

**Comment by marc0olo** [comment]
@skkw This feature is merged now in the node, right?

**Comment by gorbak25** [comment]
@skkw What's the status on persisting the events exposed by the node?

**Comment by skkw** [comment]
This is now implemented in release 1.0.4.
